File: app/src/main/java/blue/koenig/kingsfinances/view/FinanceView.java
package blue.koenig.kingsfinances.view;

import com.koenig.commonModel.User;
import com.koenig.commonModel.finance.BankAccount;
import com.koenig.commonModel.finance.StandingOrder;

import java.util.List;

import blue.koenig.kingsfamilylibrary.view.family.FamilyView;
import blue.koenig.kingsfinances.model.calculation.StatisticEntryDeprecated;

/**
 * Created by Thomas on 18.10.2017.
 */

public interface FinanceView extends FamilyView {

    void setFamilyMembers(List<User> members);

    void showStandingOrders(List<StandingOrder> standingOrders);

    void updateBankAccounts(List<BankAccount> bankAccounts);

    void updateAssets(List<StatisticEntryDeprecated> assets);

}

File: app/src/main/java/blue/koenig/kingsfinances/view/FinanceView.kt
package blue.koenig.kingsfinances.view

import blue.koenig.kingsfamilylibrary.view.family.FamilyView
import com.koenig.commonModel.User
import com.koenig.commonModel.finance.BankAccount
import com.koenig.commonModel.finance.StandingOrder
import com.koenig.commonModel.finance.statistics.MonthStatistic
import org.joda.time.YearMonth

/**
 * Created by Thomas on 18.10.2017.
 */

interface FinanceView : FamilyView {

    fun setFamilyMembers(members: List<User>)

    fun showStandingOrders(standingOrders: List<StandingOrder>)

    fun updateBankAccounts(bankAccounts: List<BankAccount>)

    fun updateAssets(assets: Map<YearMonth, MonthStatistic>)

}
